### ISSUE TITLE
[cmake] Allow configuring SDL_image with options whose 3rd party dependencies are not available

### ISSUE THREAD

**Issue by madebr** [libsdl-org/SDL_image#413]
Currently, CMake configuration fails when a 3rd party dependency is not available.
Let's change this into a soft error and just disable that option.

At the same time, introduce a strict option that fails the build when a dependency could not be fulfilled.

This has been done before in SDL_mixer: https://github.com/libsdl-org/SDL_mixer/commit/2414c440607d9c5c051dee8988699bcb7d97865e

**Comment by slouken** [comment]
Yes please! This should be done for SDL3 and not SDL2, as we did for SDL_mixer.
